Automattic/newspack-plugin
# feat: (IA) dash brand header and <BoxContrast /> component

### All Submissions:


### Changes proposed in this Pull Request:

Adds brand header component to new Newspack Dashboard. Title text and background-color are derived dynamically from:
- Title = Site name
- Background color = Site Design > Header Background Color

This PR also introduces a new reusable component named `<BoxContrast />`:
* Allows passing in a hexadecimal color value which is evaluated to for luminescence.
* The relative luminescence will determine if white or black color should be used for best contrast.
* Hex color can be applied to `color` or `background-color` css props.
* Allows passing in text, or react component, to include within the box.

To aid in local ts testing I have added `npm run tsc` to create a typescript-compiler watch instance.

### How to test the changes in this Pull Request:

1. Checkout this branch and build assets i.e. git `checkout feat/ia-dash-sections && npm run build`
2. Navigate to - /wp-admin/admin.php?page=newspack#/
3. Observe that Site Name and Background Color values have been applied to header.
4. Update Site Name and Background Color values and confirm they reflect in the header.

Below examples ("Newspack" is Site Name):

![Screenshot 2024-03-28 at 11 52 01](https://github.com/Automattic/newspack-plugin/assets/3676975/a080b80b-6ac4-4b53-a481-3ce89614c662)
![Screenshot 2024-03-28 at 11 52 54](https://github.com/Automattic/newspack-plugin/assets/3676975/6d710c3a-5f08-400e-ad90-954458114c2c)
![Screenshot 2024-03-28 at 11 54 07](https://github.com/Automattic/newspack-plugin/assets/3676975/3d8a1c46-367f-4767-bd50-103123757fec)
![Screenshot 2024-03-28 at 11 54 27](https://github.com/Automattic/newspack-plugin/assets/3676975/21efd7ff-81f4-43f9-8af6-177f4e955aca)


### Other information:
